src/service.rs

```diff
@@ -0,0 +1,278 @@
+use std::ffi::OsString;
+use std::path::PathBuf;
+
+use lazy_static::lazy_static;
+use structopt::StructOpt;
+
+use crate::cli::handle_config_errors;
+use crate::config;
+
+macro_rules! win_error {
+    ($msg:expr, $e:ident) => {
+       match $e {
+           windows_service::Error::Winapi(error) => error!(message=$msg, %error),
+           _ => error!(message=$msg, %$e)
+       }
+    };
+}
+
+lazy_static! {
+    pub static ref DEFAULT_SERVICE_NAME: &'static str = built_info::PKG_NAME;
+}
+
+#[derive(StructOpt, Debug)]
+#[structopt(rename_all = "kebab-case")]
+pub struct Opts {
+    #[structopt(subcommand)]
+    sub_command: Option<SubCommand>,
+}
+
+#[derive(StructOpt, Debug)]
+#[structopt(rename_all = "kebab-case")]
+struct InstallOpts {
+    /// The name of the service to install.
+    #[structopt(long)]
+    name: Option<String>,
+
+    /// The display name to be used by interface programs to identify the service like Windows Services App
+    #[structopt(long)]
+    display_name: Option<String>,
+
+    /// The configuration files that will be used by the service.
+    /// If no configuration file is specified, will target default configuration file.
+    #[structopt(name = "config", short, long)]
+    config_paths: Vec<PathBuf>,
+}
+
+impl InstallOpts {
+    fn service_info(&self) -> ServiceInfo {
+        let service_name = self.name.as_deref().unwrap_or(*DEFAULT_SERVICE_NAME);
+        let display_name = self.display_name.as_deref().unwrap_or("Vector Service");
+        let description = built_info::PKG_DESCRIPTION;
```

Comment:
I unfortunately did not see this get set?

![image](https://user-images.githubusercontent.com/130903/90795763-502f5200-e2c3-11ea-9d51-24c129b78fe4.png)